Assignment3.java

```diff
@@ -0,0 +1,45 @@
+import java.util.*;
+
+public class Assignment3 {
+    private Set<String> setOfWords = new HashSet<>();
+
+    public List<String> findAllWords(int rowNumber, int columnNumber, char[][] grid, MyDictionary dictionary) {
+        boolean[][] visited = new boolean[rowNumber][columnNumber];
+        String s = "";
+
+        for (int i = 0; i < rowNumber; i++) {
+            for (int j = 0; j < columnNumber; j++) {
+                findWordsStartingWithSpecificChar(grid, i, j, visited, s, dictionary);
+                visited = new boolean[rowNumber][columnNumber];
+            }
+        }
+
+        List<String> words = new ArrayList<>();
+        words.addAll(setOfWords);
+
+        return words;
+    }
+
+    private void findWordsStartingWithSpecificChar(char[][] grid, int row, int column, boolean[][] visited, String s,
+                                                   MyDictionary dictionary) {
+        if (!dictionary.isPrefix(s + grid[row][column])) {
```

Comment:
Don't think of this as a Java file that is going to compile. Just think of it as you wrote this code in Google Docs, like in the interview. In the pdf, it says that the dictionary and its methods are given.